subworkflows/local/bin_qc.nf

```diff
@@ -0,0 +1,182 @@
+include { BUSCO_SUMMARY                     } from '../../modules/local/busco_summary'
+include { CHECKM_QA                         } from '../../modules/nf-core/checkm/qa/main'
+include { CHECKM_LINEAGEWF                  } from '../../modules/nf-core/checkm/lineagewf/main'
+include { CHECKM2_PREDICT                   } from '../../modules/nf-core/checkm2/predict/main'
+include { COMBINE_TSV as COMBINE_BINQC_TSV  } from '../../modules/local/combine_tsv'
+include { GUNC_DOWNLOADDB                   } from '../../modules/nf-core/gunc/downloaddb/main'
+include { GUNC_RUN                          } from '../../modules/nf-core/gunc/run/main'
+include { GUNC_MERGECHECKM                  } from '../../modules/nf-core/gunc/mergecheckm/main'
+
+
+workflow BIN_QC {
+    take:
+    ch_bins        // [ [ meta] , fasta           ], input bins (mandatory)
+    ch_checkm_db   // [ db                        ], presupplied CheckM database (optional)
+    ch_checkm2_db  // [ [meta]  , db              ], presupplied CheckM2 database (optional)
+    ch_busco_db    // [ [meta]  , db              ], presupplied BUSCO database (optional)
+    ch_gunc_db     // [ db                        ], presupplied GUNC database (optional)
+
+    main:
+    qc_summary = []
+    ch_input_bins_for_qc = ch_bins.transpose()
+    ch_versions = Channel.empty()
+    ch_multiqc_files = Channel.empty()
+
+
+    if (params.binqc_tool == "busco") {
+        /*
+         * BUSCO
+         */
+        if (!ch_busco_db.isEmpty()) {
+            if (ch_busco_db.extension in ['gz', 'tgz']) {
+                // Expects to be tar.gz!
+                BUSCO_DB_PREPARATION(ch_busco_db)
+                ch_db_for_busco = BUSCO_DB_PREPARATION.out.db.map { meta, db ->
+                    [[id: meta, lineage: 'Y'], db]
+                }
+            }
+            else if (ch_busco_db.isDirectory()) {
+                // Set meta to match expected channel cardinality for BUSCO
+                ch_db_for_busco = Channel
+                    .of(ch_busco_db)
+                    .collect { db ->
+                        def basename = db.getBaseName()
+                        def lineage = basename.contains('odb10') ? 'Y' : 'N'
+                        [[id: basename, lineage: lineage], db]
+                    }
+            }
+        }
+        else {
+            // Set BUSCO database to empty to allow for --auto-lineage
+            ch_db_for_busco = Channel
+                .of([[lineage: ''], []])
+                .collect()
+        }
+
+        if (params.save_busco_db) {
+            // publish files downloaded by Busco
+            ch_downloads = BUSCO.out.busco_downloads
+                .groupTuple()
+                .map { _lin, downloads -> downloads[0] }
+                .toSortedList()
+                .flatten()
+            BUSCO_SAVE_DOWNLOAD(ch_downloads)
+        }
+
+        BUSCO(ch_input_bins_for_qc, ch_db_for_busco)
+
+        BUSCO_SUMMARY(
```

Comment:
Missing versions file being mixed into `ch_versions` for this process, the SAVE_DOWNLOAD doesn't have one, although given that process is entirely empty, shouldn't that be replaced by a `publishDir` specification in `modules.conf` :thinking: 

(I know you didn't write this, but might be worth fixing this now :) )

Reply:
By the comment on that module, the problem with simply uisng `publishDIr` is that the same files would get written multiple times. I couldn't think of a more elegant solution, so I will add the version, unless you have a better idea.